src/python/pants/rules/core/filedeps.py

```diff
@@ -0,0 +1,39 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from pex.orderedset import OrderedSet
+
+from pants.engine.addressable import BuildFileAddresses
+from pants.engine.console import Console
+from pants.engine.legacy.graph import TransitiveHydratedTargets
+from pants.engine.rules import console_rule
+from pants.engine.selectors import Select, Get
+
+
+@console_rule('filedeps', [Select(Console), Select(BuildFileAddresses)])
+def file_deps(console, addresses):
+  """List all source and BUILD files a target transitively depends on.
+
+  Files are listed with relative paths and any BUILD files implied in the transitive closure of
+  targets are also included.
+  """
+
+  uniq_set = OrderedSet()
+
+  transitive_hydrated_targets = yield Get(
+    TransitiveHydratedTargets,
+    BuildFileAddresses,
+    addresses,
+  )
+
+  for tht in transitive_hydrated_targets.closure:
```

Comment:
`tht` is a misleading name, because these are actually `HydratedTargets` - maybe call this `target` or `hydrated_target` (or maybe `ht`, but a longer name is probably better)